examples/src/test/kotlin/EnterpriseAuthTest.kt

```diff
@@ -113,6 +117,88 @@ internal class EnterpriseAuthTest {
         val mongoClient = MongoClient.create(connectionString)
         // :snippet-end:
     }
+
+    fun oidcAzureConnectionString() = runBlocking {
+        // :snippet-start: oidc-azure-connection-string
+        val connectionString = ConnectionString(
+            "mongodb://<username>@<hostname>:<port>/?" +
+                "?authMechanism=MONGODB-OIDC" +
+                "&authMechanismProperties=ENVIRONMENT:azure,TOKEN_RESOURCE:<percent-encoded audience>")
+        val mongoClient = MongoClient.create(connectionString)
+        // :snippet-end:
+    }
+
+    fun oidcAzureCredential() = runBlocking {
+        // :snippet-start: oidc-azure-credential
+        val credential = MongoCredential.createOidcCredential("<username>")
+            .withMechanismProperty("ENVIRONMENT", "azure")
+            .withMechanismProperty("TOKEN_RESOURCE", "<audience>")
+
+        val mongoClient = MongoClient.create(
+                MongoClientSettings.builder()
+                    .applyToClusterSettings { builder ->
+                        builder.hosts(listOf(ServerAddress("<hostname>", PORT)))
+                    }
+                .credential(credential)
+                .build())
+        // :snippet-end:
+    }
+
+    fun oidcGCPConnectionString() = runBlocking {
+        // :snippet-start: oidc-gcp-connection-string
+        val connectionString = ConnectionString(
+            "mongodb://<hostname>:<port>/?" +
+                    "authMechanism=MONGODB-OIDC" +
+                    "&authMechanismProperties=ENVIRONMENT:gcp,TOKEN_RESOURCE:<percent-encoded audience>")
+        val mongoClient = MongoClient.create(connectionString)
+        // :snippet-end:
+    }
+
+    fun oidcGCPCredential() = runBlocking {
+        // :snippet-start: oidc-gcp-credential
+        val credential = MongoCredential.createOidcCredential("<username>")
+            .withMechanismProperty("ENVIRONMENT", "gcp")
+            .withMechanismProperty("TOKEN_RESOURCE", "<audience>")
+
+        val mongoClient = MongoClient.create(
+            MongoClientSettings.builder()
+                .applyToClusterSettings { builder ->
+                    builder.hosts(listOf(ServerAddress("<hostname>", PORT)))
+                }
+                .credential(credential)
+                .build())
+        // :snippet-end:
+    }
+
+    fun oidcCallback() = runBlocking {
+        // :snippet-start: oidc-callback
+        val credential = MongoCredential.createOidcCredential(null)
+            .withMechanismProperty("OIDC_CALLBACK") { Context context ->
+                val accessToken = "..."
+                OidcCallbackResult(accessToken)
+            }
+        // :snippet-end:
+    }
+
+
+    fun oidcCallbackFile() = runBlocking {
+        // :snippet-start: oidc-callback-file
+        val credential = MongoCredential.createOidcCredential(null)
+            .withMechanismProperty("OIDC_CALLBACK") { Context context ->
```

Comment:
This doesn't compile in Kotlin try: 

```suggestion
            .withMechanismProperty("OIDC_CALLBACK") { context: Context ->
```